sympy/matrices/determinant.py

```diff
@@ -0,0 +1,743 @@
+from __future__ import division, print_function
+
+from types import FunctionType
+
+from sympy.core.cache import cacheit
+from sympy.core.function import expand_mul
+from sympy.core.numbers import Float, Integer
+from sympy.core.singleton import S
+from sympy.core.symbol import _uniquely_named_symbol
+from sympy.core.sympify import sympify
+from sympy.polys import PurePoly, cancel
+from sympy.simplify import simplify as _simplify, dotprodsimp as _dotprodsimp
+
+from .common import NonSquareMatrixError
+
+from .utilities import _toselfclass, _iszero, _is_zero_after_expand_mul
+
+
+def _find_reasonable_pivot(col, iszerofunc=_iszero, simpfunc=_simplify):

```

Comment:
I would rather move this to the `utilities.py`

Reply:
You mean `sympy.utilities`? Yeah, no problem there.

Reply:
I don't see why this would go in utilities. It's not really any use outside of matrices

Reply:
I had meant by matrices.utilities.py